### PR TITLE
Set `@icon` for models and connectors

### PR DESCRIPTION
**Pull request by ven-k**
- Adds `@icon` to specify the URI, path to a PNG or SVG file, or inline SVG

    - `gui_metadata.layout` is set to provided icon
    - PNG or SVG files can be specified with
        - full path
        - filename (when it is at either env var `MTK_ICONS_DIR or , when that isn't set, at `DEPOT_PATH[1]/mtk_icons`
      These are stored as file-format-URI.
    - A URI can be passed too.
    -  An SVG can be inlined with a begin...end block
- Adds related tests

- While here, propogates the gui_metadata of sys to the extended system

### CODE REVIEW

File: test/model_parsing.jl [hunk 1]
```diff
@@ -24,6 +28,7 @@ D = Differential(t)
 @connector Pin begin
     v(t) = 0                  # Potential at the pin [V]
     i(t), [connect = Flow]    # Current flowing into the pin [A]
+    @icon "pin.png"
```
Comment (YingboMa) [inline test/model_parsing.jl:31]: Can it support relative path with `../` etc and absolute path?

File: Project.toml [hunk 2]
```diff
@@ -44,6 +44,7 @@ StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
 SymbolicIndexingInterface = "2efcf032-c050-4f8e-a9bb-153293bab1f5"
 SymbolicUtils = "d1185830-fcd6-423d-90d6-eec64667417b"
 Symbolics = "0c5d862f-8b57-4792-8d23-62f2024744c7"
+URIs = "5c2747f8-b7ea-4ff2-ba2e-563bfd36b1d4"
```
[the remark this answers is not in the document]
Comment (ven-k) [reply Project.toml:47]: CI wasn't triggered for https://github.com/ven-k/ModelingToolkit.jl/commit/28fd47764cc6a8a3b68a405d65b5b9b465dd41e1.

I've retriggered with https://github.com/SciML/ModelingToolkit.jl/pull/2183/commits/afb41104b29ec0a9f349200e360dccc9a568c739

**Comment by YingboMa** [comment]
Could you merge master back to this branch? That might help with the CI failure.

**Comment by ven-k** [comment]
It's already rebased to current master[^1].

Should I add more tests to increase the coverage?

[^1]: https://github.com/ven-k/ModelingToolkit.jl/commits/vkb/gui_metadata

**Comment by ven-k** [comment]
Locally I could reproduce SBMLToolkit test failure; but its sporadic.

**Review by ChrisRackauckas — APPROVED** [review]

**Review by YingboMa — APPROVED** [review]

**Pull request merged**